big_tests/tests/service_domain_db_SUITE.erl

```diff
@@ -0,0 +1,501 @@
+    %% Simulate MIM restart
+    service_disabled(mim()),
+    %% Only one domain is locked
+    init_with(mim(), [{<<"example.com">>, <<"cfggroup">>}], [<<"dbgroup">>, <<"cfggroup">>]),
+    service_enabled(mim()),
+    %% DB still contains data
+    {ok, #{host_type := <<"dbgroup">>, enabled := true}} =
+       select_domain(mim(), <<"example.com">>),
+    {ok, #{host_type := <<"dbgroup">>, enabled := true}} =
+       select_domain(mim(), <<"example.net">>),
+     %% Locked DB records are ignored
+    {ok, <<"cfggroup">>} = get_host_type(mim(), <<"example.com">>),
+    {ok, <<"dbgroup">>} = get_host_type(mim(), <<"example.net">>).
+
+db_events_table_gets_truncated(_) ->
+    precond(mim(), off, [], [<<"dbgroup">>]),
+    %% Configure service with a very short interval
+    service_enabled(mim(), [{event_cleaning_interval, 1}, {event_max_age, 3}]),
+    ok = insert_domain(mim(), <<"example.com">>, <<"dbgroup">>),
+    ok = insert_domain(mim(), <<"example.net">>, <<"dbgroup">>),
+    ok = insert_domain(mim(), <<"example.org">>, <<"dbgroup">>),
+    ok = insert_domain(mim(), <<"example.beta">>, <<"dbgroup">>),
+    Max = get_max_event_id(mim()),
+    true = is_integer(Max),
+    true = Max > 0,
+    %% The events table is not empty and the size of 1, eventually.
+    F = fun() -> get_min_event_id(mim()) end,
+    mongoose_helper:wait_until(F, Max, #{time_left => timer:seconds(15)}),
+    ok.
+
+db_could_sync_between_nodes(_) ->
+    precond(mim(), on, [], [<<"dbgroup">>]),
+    precond(mim2(), on, [], [<<"dbgroup">>]),
+    ok = insert_domain(mim(), <<"example.com">>, <<"dbgroup">>),
+    sync(),
+    {ok, <<"dbgroup">>} = get_host_type(mim2(), <<"example.com">>),
+    ok.
+
+db_removed_from_one_node_while_service_disabled_on_another(_) ->
+    precond(mim(), on, [], [<<"dbgroup">>]),
+    precond(mim2(), on, [], [<<"dbgroup">>]),
+    ok = insert_domain(mim(), <<"example.com">>, <<"dbgroup">>),
+    sync(),
+    {ok, <<"dbgroup">>} = get_host_type(mim2(), <<"example.com">>),
+    %% Service is disable on the second node
+    service_disabled(mim2()),
+    %% Removed from the first node
+    ok = remove_domain(mim(), <<"example.com">>, <<"dbgroup">>),
+    sync(),
+    {error, not_found} = get_host_type(mim(), <<"example.com">>),
+    {ok, <<"dbgroup">>} = get_host_type(mim2(), <<"example.com">>),
+    %% Sync is working again
+    service_enabled(mim2()),
+    {error, not_found} = get_host_type(mim2(), <<"example.com">>),
+    ok.
+
+db_inserted_from_one_node_while_service_disabled_on_another(_) ->
+    precond(mim(), on, [], [<<"dbgroup">>]),
+    precond(mim2(), on, [], [<<"dbgroup">>]),
+    %% Service is disable on the second node
+    service_disabled(mim2()),
+    ok = insert_domain(mim(), <<"example.com">>, <<"dbgroup">>),
+    %% Sync is working again
+    service_enabled(mim2()),
+    {ok, <<"dbgroup">>} = get_host_type(mim2(), <<"example.com">>),
+    ok.
+
+db_reinserted_from_one_node_while_service_disabled_on_another(_) ->
```

Comment:
yea, but if there is some static data in DB tied to the host-type, it would not be migrated.
It's also the behaviour **from the requirements**, I've just added test to document it too.